PlayFabClientSDK/source/PlayFabHttp/PlayFabHttp.cs

```diff
@@ -41,28 +41,28 @@ public class PlayFabJsonSuccess<TResult> where TResult : PlayFabResultCommon
 
     public static class PlayFabHttp
     {
-        private static readonly IPlayFabHttp _http;
-
-        static PlayFabHttp()
-        {
-            var httpInterfaceType = typeof(IPlayFabHttp);
-            var types = typeof(PlayFabHttp).GetAssembly().GetTypes();
-            foreach (var eachType in types)
-            {
-                if (httpInterfaceType.IsAssignableFrom(eachType) && !eachType.IsAbstract)
-                {
-                    _http = (IPlayFabHttp)Activator.CreateInstance(eachType.AsType());
-                    return;
-                }
-            }
-            throw new Exception("Cannot find a valid IPlayFabHttp type");
-        }
+        private static ITransportPlugin _transport;
 
         public static async Task<object> DoPost(string urlPath, PlayFabRequestCommon request, string authType, string authKey, Dictionary<string, string> extraHeaders)
         {
             if (PlayFabSettings.TitleId == null)
                 throw new Exception("You must set your titleId before making an api call");
-            return await _http.DoPost(urlPath, request, authType, authKey, extraHeaders);
+            if (_transport == null)
+                _transport = GetTransport();
+
+            return await _transport.DoPost(urlPath, request, authType, authKey, extraHeaders);
```

Comment:
Should we null check _transport?

Reply:
It's null checked just above

---
In reply to: [200494969](https://github.com/PlayFab/CSharpSDK/pull/97#discussion_r200494969) [](ancestors = 200494969)

Reply:
By updated design plugins won't be allowed to be null, i.e. PluginManager will always return a non-null instance of a plugin

---
In reply to: [200499576](https://github.com/PlayFab/CSharpSDK/pull/97#discussion_r200499576) [](ancestors = 200499576,200494969)